spec/services/facility_stats_service_spec.rb

```diff
@@ -0,0 +1,116 @@
+  let!(:small_facility1) { create(:facility, name: 'small1', facility_size: 'small', facility_group: facility_group) }
+  let!(:small_facility2) { create(:facility, name: 'small2', facility_size: 'small', facility_group: facility_group) }
+  let!(:medium_facility1) { create(:facility, name: 'medium1', facility_size: 'medium', facility_group: facility_group) }
+  let!(:medium_facility2) { create(:facility, name: 'medium2', facility_size: 'medium', facility_group: facility_group) }
+  let!(:large_facility1) { create(:facility, name: 'large1', facility_size: 'large', facility_group: facility_group) }
+  let!(:large_facility2) { create(:facility, name: 'large2', facility_size: 'large', facility_group: facility_group) }
+  let(:all_facilities) { [small_facility1, small_facility2, medium_facility1, medium_facility2, large_facility1, large_facility2] }
+  let(:period) { Period.month(today) }
+
+  # this is intended to create a semi-randomized assortment of data that covers most cases
+  before :each do
+    small_controlled = nil
+    small_uncontrolled = nil
+    medium_controlled = nil
+    medium_uncontrolled = nil
+    large_controlled = nil
+    large_uncontrolled = nil
+    Timecop.freeze(six_months_ago) do
+      large_controlled = create_list(:patient, 3, full_name: 'large_controlled', assigned_facility: large_facility1)
+    end
+    Timecop.freeze(five_months_ago) do
+      small_controlled = create_list(:patient, 2, full_name: 'small_controlled', assigned_facility: small_facility1, registration_user: supervisor)
+      small_uncontrolled = create_list(:patient, 1, full_name: 'small_uncontrolled', assigned_facility: small_facility2, registration_user: supervisor)
+      medium_uncontrolled = create_list(:patient, 2, full_name: 'medium_uncontrolled', assigned_facility: medium_facility1)
+      large_uncontrolled = create_list(:patient, 1, full_name: 'large_uncontrolled', assigned_facility: large_facility2)
+
+      medium_uncontrolled.each { |patient| create(:blood_pressure, :hypertensive, patient: patient, facility: patient.assigned_facility) }
+      large_controlled.each { |patient| create(:blood_pressure, :under_control, patient: patient, facility: patient.assigned_facility) }
+    end
+    Timecop.freeze(four_months_ago) do
+      medium_controlled = create_list(:patient, 2, full_name: 'medium_uncontrolled', assigned_facility: medium_facility1)
+
+      medium_controlled.each { |patient| create(:blood_pressure, :under_control, patient: patient, facility: patient.assigned_facility) }
+      small_controlled.each { |patient| create(:blood_pressure, :under_control, patient: patient, facility: patient.assigned_facility, user: supervisor) }
+      small_uncontrolled.each { |patient| create(:blood_pressure, :hypertensive, patient: patient, facility: patient.assigned_facility, user: supervisor) }
+      large_uncontrolled.each { |patient| create(:blood_pressure, :hypertensive, patient: patient, facility: patient.assigned_facility) }
+    end
```

Comment:
This is a _ton_ of test data that most of the tests don't use - I would recommend creating just the test data you need inside the specific specs. It will make for more clear specs and will also be much faster.